solidity/contracts/bridge/VendingMachine.sol

```diff
@@ -0,0 +1,313 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../token/TBTCToken.sol";
+import "../token/IReceiveApproval.sol";
+import "../GovernanceUtils.sol";
+
+/// @title TBTC v2 Vending Machine
+/// @notice The Vending Machine is the owner of TBTC v2 token and can mint
+///         TBTC v2 tokens in 1:1 ratio from TBTC v1 tokens with TBTC v1
+///         deposited in the contract as collateral. TBTC v2 can be
+///         unminted back to TBTC v1 with or without a fee - fee parameter is
+///         controlled by the Governance. This implementation acts as a bridge
+///         between TBTC v1 and TBTC v2 token, allowing to mint TBTC v2 before
+///         the system is ready and fully operational without sacrificing any
+///         security guarantees and decentralization of the project.
+///         Vending Machine can be upgraded in a two-step, governance-controlled
+///         process. The new version of the Vending Machine will receive the
+///         ownership of TBTC v2 token and entire TBTC v1 balance stored as
+///         collateral. It is expected that this process will be executed before
+///         the v2 system launch. There is an optional unmint fee with a value
+///         that can be updated in a two-step, governance-controlled process.
+///         All governable parameters are controlled by two roles: update
+///         initiator and finalizer. There is a separate initiator role for
+///         unmint fee update and vending machine upgrade. The initiator
+///         proposes the change by initiating the update and the finalizer
+///         (contract owner) may approve it by finalizing the change after the
+///         governance delay passes.
+contract VendingMachine is Ownable, IReceiveApproval {
+    using SafeERC20 for IERC20;
+    using SafeERC20 for TBTCToken;
+
+    /// @notice The time delay that needs to pass between initializing and
+    ///         finalizing update of any governable parameter in this contract.
+    uint256 public constant GOVERNANCE_DELAY = 7 days;
```

Comment:
Correct. The pause I meant was not for an emergency case, it was for the case when minting has to be temporarily disabled with an advance notice so that v1 can be safely winded down.